rundeckapp/grails-app/controllers/rundeck/controllers/StorageController.groovy

```diff
@@ -268,19 +269,13 @@ class StorageController extends ControllerBase{
         }
         AuthContext authContext = getAuthContextForPath(session.subject, storageParams.resourcePath)
         def resourcePath = storageParams.resourcePath
-        def valid=false
-        withForm {
-            valid=true
-        }.invalidToken{
+
+        withForm { }.invalidToken{
             flash.errorCode= 'request.error.invalidtoken.message'
             return redirect(controller: 'menu', action: 'storage', params: [project: params.project])
         }
-        if(!valid){
-            return
-        }
+
```

Comment:
the `valid` variable is still necessary. without that check, execution of `keyStorageUpload` method will not return.  the `return` inside the closure does not return from the outer method